### DISCUSSION TITLE
It works great.

### DISCUSSION THREAD

**Discussion by elephantpanda** [microsoft/onnxruntime-genai#812] [General]
Thanks for this genai functionality. I tried it with phi-3 directml int4 and it works great. It works very fast with my P5000 GPU and also works not too slowly on my integrated Core i3 Intel GPU.

One question I have is that I notice that the default functionality of the python script doesn't remember your previous answer. It would be nice if there was a setting where it didn't forget all your previous answers.

Also, nice would be if you could add your own global prompt such as to tell it what kind of personality to have.

One thing that is a bit annoying is that phi-3 sometimes refuses to answer some questions but that is probably more to do with how it was fine-tuned than anything else.

Overall the python script is a nice way to test out the LLMs. I will also be trying out the C# code to incorporate this into a game.

**Comment by kunal-vaishnavi** [comment]
> One question I have is that I notice that the default functionality of the python script doesn't remember your previous answer. It would be nice if there was a setting where it didn't forget all your previous answers.
>
> Also, nice would be if you could add your own global prompt such as to tell it what kind of personality to have.

Support for continuous decoding and system prompt pre-processing has been merged [here](https://github.com/microsoft/onnxruntime-genai/pull/867).
